src/routes/manage_job/post.rs

```diff
@@ -0,0 +1,294 @@
+	// Additional arguments are optional, but can be used to access context
+	// provided via [`JobRegistry::set_context`].
+) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+	let task_input: TaskInput = current_job.json()?.unwrap();
+
+	// Retry each future twice, to avoid grey-listing.
+	if let Ok((response, _)) = retry(&task_input.input, RetryOption::Direct, 2).await {
+		log::debug!(
+			target:"reacher",
+			"Succeeded [email={}] for [job={}]",
+			task_input.job_id,
+			task_input.input.to_emails[0]
+		);
+
+		let rec = sqlx::query!(
+			r#"
+			INSERT INTO email_results (job_id, result)
+			VALUES ($1, $2)
+			"#,
+			task_input.job_id,
+			serde_json::json!(response)
+		)
+		// TODO: This is a simplified solution and will work when
+		// the task queue and email results tables are in the same
+		// database. Keeping them in separate database will require
+		// some custom logic on the job registry side
+		// https://github.com/Diggsey/sqlxmq/issues/4
+		.fetch_optional(current_job.pool())
+		.await
+		.map_err(|e| {
+			log::debug!(
+				target:"reacher",
+				"Failed to write [email={}] result to db for [job={}]",
+				task_input.job_id,
+				task_input.input.to_emails[0]
+			);
+		});
+	} else {
+		log::debug!(
+			target:"reacher",
+			"Failed [email={}] for [job={}]",
+			task_input.job_id,
+			task_input.input.to_emails[0]
+		);
+	}
+
+	current_job.complete().await?;
+
+	Ok(())
+}
+
+/// Retry the check ciee_check_email function, in particular to avoid
+/// greylisting.
+#[async_recursion]
+async fn retry(
+	input: &CheckEmailInput,
+	retry_option: RetryOption,
+	count: usize,
+) -> Result<(CheckEmailOutput, RetryOption), CheckEmailError> {
+	log::debug!(
+		target:"reacher",
+		"[email={}] Checking with retry option {}, attempt #{}",
+		input.to_emails[0],
+		retry_option,
+		count,
+	);
+
+	let result = ciee_check_email(input)
+		.await
+		.pop()
+		.expect("Input contains one email, so does output. qed.");
+
+	log::debug!(
+		target:"reacher",
+		"[email={}] Got result with retry option {}, attempt #{}, is_reachable={:?}",
+		input.to_emails[0],
+		retry_option,
+		count,
+		result.is_reachable
+	);
+
+	// If we get an unknown error, log it.
+	known_errors::log_unknown_errors(&result, retry_option);
+
+	if result.is_reachable == Reachable::Unknown {
+		if count <= 1 {
+			Err(CheckEmailError::Unknown((result, retry_option)))
+		} else {
+			retry(input, retry_option, count - 1).await
+		}
+	} else {
+		Ok((result, retry_option))
+	}
+}
+
+/// handles input, creates db entry for job and tasks for verification
+async fn create_bulk_request(
+	body: CreateBulkRequestBody,
+	conn_pool: Pool<Postgres>,
+) -> Result<impl warp::Reply, warp::Rejection> {
+	// create job entry
+	let rec = sqlx::query!(
+		r#"
+		INSERT INTO bulk_jobs (total_records)
+		VALUES ($1)
+		RETURNING id
+		"#,
+		body.input.len() as i32
+	)
+	.fetch_one(&conn_pool)
+	.await
+	.map_err(|e| {
+		log::error!(
+			target:"reacher/v0/bulk/",
```

Comment:
```suggestion
			target:"reacher",
```

let's use target:"reacher" everywhere